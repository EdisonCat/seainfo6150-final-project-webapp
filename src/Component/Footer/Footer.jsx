import React from "react";
import styles from "./Footer.module.css";
import { useHistory} from "react-router-dom";
export default function Footer() {
    const history = useHistory();
    const handleHome = () => {
        history.push({pathname: "/"});
    }
    return (
        <div className={styles.container}>
            <ul>
                <li onClick={() => handleHome()} className={styles.link}>Marvel Fandom</li>
                <li>made by&nbsp;<a className={styles.link} target="_blank" rel="noopener noreferrer" href="https://edisonguo.tech">Edison</a></li>
            </ul>
        </div>
    );
}