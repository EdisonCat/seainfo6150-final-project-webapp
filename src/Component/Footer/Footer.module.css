.container {
    background-color: #333;
    color: white;
    position: relative;
    bottom: 0;
    width: 100%;
    border-top: 1px solid grey;
}
.container ul {
    list-style: none;
}
@import url('https://fonts.googleapis.com/css2?family=Work+Sans:wght@300&display=swap');
.container li {
    width: 100%;
    display: flex;
    justify-content: center;
	font-family: 'Work Sans', sans-serif;
}
.link {
    text-decoration: underline;
    text-decoration-color: #333;
    transition: 0.5s;
    cursor: pointer;
    outline: none;
    color: white;
}
.link:hover {
    text-decoration-color: red;
}