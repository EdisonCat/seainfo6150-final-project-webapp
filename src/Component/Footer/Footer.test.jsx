import React from "react";
import Footer from "./Footer.jsx";

it("renders the Footer component correctly", () => {
	// eslint-disable-next-line no-undef
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});
