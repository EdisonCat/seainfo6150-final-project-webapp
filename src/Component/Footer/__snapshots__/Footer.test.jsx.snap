// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Footer component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <ul>
      <li
        class="link"
      >
        Marvel Fandom
      </li>
      <li>
        made byÂ 
        <a
          class="link"
          href="https://edisonguo.tech"
          rel="noopener noreferrer"
          target="_blank"
        >
          Edison
        </a>
      </li>
    </ul>
  </div>
</div>
`;
